scenes/globals/pause.gd

```diff
@@ -0,0 +1,91 @@
+extends Node
+
+enum System {
+	PLAYER_INPUT,
+	GAME
+	# For now is just these two but in the future we could have: Physics, Enemies, AI, etc
+}
+
+# Private vars
+var _pause_requests: Dictionary[System, Array] = {}
+
+# This should be an array of systems but it throws
+# a runtime error when using the filter function, so for now just "Array"
+var _paused_systems: Array = []
+
+
+# Public interface
+func pause_system(system: System, node: Node) -> void:
+	_add_request(system, node)
+
+
+func unpause_system(system: System, node: Node) -> void:
+	_clear_request(system, node)
+
+	_refresh_systems()
+
+
+func is_paused(system: System) -> bool:
+	return system in paused_systems()
+
+
+func paused_systems() -> Array:
+	return _paused_systems
+
+
+# Private methods
+func _exists_request_for_system(a_system: System) -> bool:
+	return !_pause_requests.get_or_add(a_system, []).is_empty()
```

Comment:
The `!` negation is deprecated in favor of `not`. Also why not just check if the key is in the Dictionary? Which also makes the function so trivial that `a_system not in _pause_requests` can be used directly when this function is needed.
```suggestion
	return a_system not in _pause_requests
```

Reply:
Actually not, since the key is not removed when the value array is emptied.

Reply:
Then in that case I prefer that we define the variable with empty arrays, rather than using `get_or_add(system, [])` everywhere:

```
var _pause_requests: Dictionary[System, Array] = {
        System.PLAYER_INPUT: [],
        System.GAME: [],
}
```